core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -571,7 +572,7 @@ final SearchContext createContext(ShardSearchRequest request, @Nullable Engine.S
                     QueryParseContext queryParseContext = new QueryParseContext(indicesService.getIndicesQueryRegistry());
                     queryParseContext.reset(parser);
                     queryParseContext.parseFieldMatcher(parseFieldMatcher);
-                    parseSource(context, SearchSourceBuilder.fromXContent(parser, queryParseContext, aggParsers, suggesters));
+                    parseSource(context, SearchSourceBuilder.fromXContent(queryParseContext, aggParsers, suggesters));
```

Comment:
I agree but would prefer to do such renamings in a separate issue, also to avoid to many conflicts.